File: src/modifiers/textFieldStyle.ts
import { createModifier } from './createModifier';

export type TextFieldStyle = 'automatic' | 'plain' | 'roundedBorder';

/// SwiftUI `.textFieldStyle(_:)`. Sets the visual style applied to
/// `TextField`s within the view. `'automatic'` defers to the platform
/// default.
export function textFieldStyle(style: TextFieldStyle = 'automatic') {
  return createModifier('textFieldStyle', { style });
}

File: apple/Sources/ReactNativeWatchOS/Modifiers/TextFieldStyle.swift
import SwiftUI

/// SwiftUI `.textFieldStyle(_:)`. watchOS only ships `DefaultTextFieldStyle`
/// (`.automatic`); `.plain` and `.roundedBorder` are unavailable on watchOS,
/// so every value maps to `.automatic`. The `style` param is accepted for
/// cross-platform parity but has no other effect here.
enum RNWTextFieldStyleModifier {
    @MainActor
    static func register(into r: RNWModifierRegistry) {
        r.register("textFieldStyle") { view, _, _ in
            AnyView(view.textFieldStyle(.automatic))
        }
    }
}
